### PR TITLE
NAS-112123 / 21.10 / create cluster API test framework

### PR DESCRIPTION
**Pull request by yocalebo**
This adds a new area `middlewared/tests/cluster` that will be used for writing cluster specific API tests. This commit is the first part needed before actually writing the api tests.

1. create `run-cluster-tests.py` that will be called by an internal Jenkins CI job
2. store the various cluster related config in `config.py` so when changes are made we don't have to rewrite everything
3. write wrapper methods for making REST and/or Websocket calls in `utils.py`
4. add `--initialize-cluster` argument to `run-cluster-tests.py` that is responsible for "initializing" all the cluster nodes on each Jenkins CI job since the cluster config is wiped (by design) when the Jenkins job is run.
5. adds wrapper methods for running ssh commands on the nodes in the cluster (untested)

### CODE REVIEW

**Review by rick-mesta — APPROVED** [review]
LGTM 😁

File: tests/cluster/config.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+DOMAIN = '.tn.ixsystems.net'
+NET_INFO = {
```
Comment (ericbsd) [inline tests/cluster/config.py:2]: @pkgdemon and @yocalebo should this be source somewhere else?

File: tests/cluster/config.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    'CLUSTER': {'ip': '10.234.64.71', 'fqdn': f'tn-kvm01{DOMAIN}'},
+    'NODE_A': {'ip': '10.234.65.10', 'fqdn': f'tn-kvm01-nodea{DOMAIN}'},
+    'NODE_B': {'ip': '10.234.65.11', 'fddn': f'tn-kvm01-nodeb{DOMAIN}'},
+    'NODE_C': {'ip': '10.234.65.12', 'fqdn': f'tn-kvm01-nodec{DOMAIN}'},
+    'NETMASK': 22,
+    'INTERFACE': 'enp1s0',
+    'DEFGW': '10.234.64.1',
+    'DNS1': '10.231.0.2',
+    'DOMAIN': DOMAIN,
+}
```
Comment (ericbsd) [inline tests/cluster/config.py:12]: @yocalebo and @pkgdemon, should this be sourced privately?

Comment (yocalebo) [reply tests/cluster/config.py:12]: @ericbsd I see no harm with this being public 🤷

**Comment by yocalebo** [comment]
I pushed changes to our private repo for sourcing the cluster configuration information and then updated the PR accordingly.

**Pull request merged**